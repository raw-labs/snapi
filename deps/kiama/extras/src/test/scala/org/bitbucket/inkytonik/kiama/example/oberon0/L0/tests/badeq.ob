MODULE badeq;

VAR
    avar : INTEGER;
    bvar : BOOLEAN;

BEGIN
    bvar := avar = bvar;
    bvar := bvar # avar;
    bvar := bvar = TRUE;
    bvar := bvar # FALSE
END badeq.
