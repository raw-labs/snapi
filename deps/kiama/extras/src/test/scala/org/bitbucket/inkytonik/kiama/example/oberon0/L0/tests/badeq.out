* _pp.ob
MODULE badeq;

    VAR
        avar : INTEGER;
        bvar : BOOLEAN;

BEGIN
    bvar := avar = bvar;
    bvar := bvar # avar;
    bvar := bvar = TRUE;
    bvar := bvar # FALSE
END badeq.
[3]* errors
[3]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L0/tests/badeq.ob:8:20:error: got BOOLEAN, but expected INTEGER
[3]    bvar := avar = bvar;
[3]                   ^
[3]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L0/tests/badeq.ob:9:13:error: got BOOLEAN, but expected INTEGER
[3]    bvar := bvar # avar;
[3]            ^
[3]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L0/tests/badeq.ob:10:13:error: got BOOLEAN, but expected INTEGER
[3]    bvar := bvar = TRUE;
[3]            ^
[3]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L0/tests/badeq.ob:10:20:error: got BOOLEAN, but expected INTEGER
[3]    bvar := bvar = TRUE;
[3]                   ^
[3]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L0/tests/badeq.ob:11:13:error: got BOOLEAN, but expected INTEGER
[3]    bvar := bvar # FALSE
[3]            ^
[3]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L0/tests/badeq.ob:11:20:error: got BOOLEAN, but expected INTEGER
[3]    bvar := bvar # FALSE
[3]                   ^
